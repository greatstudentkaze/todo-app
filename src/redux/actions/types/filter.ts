export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY';
