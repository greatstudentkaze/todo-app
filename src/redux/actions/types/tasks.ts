export const SET_TASKS = 'SET_TASKS';
export const ADD_TASK = 'ADD_TASK';
export const REMOVE_TASK = 'REMOVE_TASK';
export const CHANGE_TASK_TITLE = 'CHANGE_TASK_TITLE';
export const TOGGLE_TASK = 'TOGGLE_TASK';
