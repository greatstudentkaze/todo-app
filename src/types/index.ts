export type TaskItem = {
  id: string,
  title: string,
  completed: boolean,
};
