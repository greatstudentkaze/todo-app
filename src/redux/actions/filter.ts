import { SET_SEARCH_QUERY } from './types/filter';

type SearchQuery = string;

interface ISetSearchQueryAction {
  type: typeof SET_SEARCH_QUERY,
  payload: string
}

export const setSearchQuery = (value: SearchQuery): ISetSearchQueryAction => ({
  type: SET_SEARCH_QUERY,
  payload: value
});

export type FilterActionType = ISetSearchQueryAction;
