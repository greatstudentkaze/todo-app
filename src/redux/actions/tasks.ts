import { ADD_TASK, CHANGE_TASK_TITLE, REMOVE_TASK, SET_TASKS, TOGGLE_TASK } from './types/tasks';
import { TaskItem, Tasks } from '../../types';

interface ISetTasksAction {
  type: typeof SET_TASKS,
  payload: Tasks,
}

interface IAddTaskAction {
  type: typeof ADD_TASK,
  payload: TaskItem,
}

interface IRemoveTaskAction {
  type: typeof REMOVE_TASK,
  payload: string,
}

interface IChangeTaskTitleAction {
  type: typeof CHANGE_TASK_TITLE,
  payload: {
    id: string,
    title: string
  },
}

interface IToggleTaskAction {
  type: typeof TOGGLE_TASK,
  payload: string,
}

export type TasksActionType = ISetTasksAction | IAddTaskAction | IRemoveTaskAction | IChangeTaskTitleAction | IToggleTaskAction;

export const addTask = (task: TaskItem): TasksActionType => ({
  type: ADD_TASK,
  payload: task,
});

export const removeTask = (id: string): TasksActionType => ({
  type: REMOVE_TASK,
  payload: id,
});

export const changeTaskTitle = (id: string, title: string): TasksActionType => ({
  type: CHANGE_TASK_TITLE,
  payload: {
    id,
    title,
  },
});

export const toggleTask = (id: string): TasksActionType => ({
  type: TOGGLE_TASK,
  payload: id,
});
