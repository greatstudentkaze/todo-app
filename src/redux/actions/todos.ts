import { ADD_TODO, CHANGE_TODO_TITLE, REMOVE_TODO, SET_TODOS, TOGGLE_TODO } from '../types/todos';
import { ToDoItem } from '../../types';

interface SetToDosAction {
  type: typeof SET_TODOS,
  payload: ToDoItem[],
}

interface AddToDoAction {
  type: typeof ADD_TODO,
  payload: ToDoItem,
}

interface RemoveToDoAction {
  type: typeof REMOVE_TODO,
  payload: string,
}

interface ChangeToDoTitleAction {
  type: typeof CHANGE_TODO_TITLE,
  payload: {
    id: string,
    title: string
  },
}

interface ToggleToDoAction {
  type: typeof TOGGLE_TODO,
  payload: string,
}

export const addToDo = (todo: ToDoItem) => ({
  type: ADD_TODO,
  payload: todo,
});

export const removeToDo = (id: string) => ({
  type: REMOVE_TODO,
  payload: id,
});

export const changeToDoTitle = (id: string, title: string) => ({
  type: CHANGE_TODO_TITLE,
  payload: {
    id,
    title,
  },
});

export const toggleToDo = (id: string) => ({
  type: TOGGLE_TODO,
  payload: id,
});

export type ToDosActionType = SetToDosAction | AddToDoAction | RemoveToDoAction | ChangeToDoTitleAction | ToggleToDoAction;
