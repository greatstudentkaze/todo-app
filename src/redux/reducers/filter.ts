import { SET_SEARCH_QUERY } from '../types/filter';
import { FilterActionType } from '../actions/filter';

const initialState = {
  searchQuery: '',
};

export const filter = (state = initialState, action: FilterActionType) => {
  switch (action.type) {
    case SET_SEARCH_QUERY:
      return {
        ...state,
        searchQuery: action.payload,
      }
    default:
      return state;
  }
};
