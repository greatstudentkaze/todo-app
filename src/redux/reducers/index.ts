import { combineReducers } from 'redux';

import { filter } from './filter';
import { tasks } from './tasks';

const rootReducer = combineReducers({
  filter,
  tasks,
});

export default rootReducer;
