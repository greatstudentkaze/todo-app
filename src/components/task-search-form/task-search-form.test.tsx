import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { Store } from 'redux';

import TaskSearchForm from './index';
import { setSearchQuery } from '../../redux/actions/filter';
import { RootState } from '../../redux/reducers';
import { Tasks } from '../../types';

const tasks: Tasks = [];

const filter = {
  searchQuery: '',
};

const initialState: RootState = {
  tasks,
  filter,
};

const mockStore = configureStore([]);

describe('TaskSearchForm component', () => {
  let store: Store;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <TaskSearchForm />
      </Provider>
    );
  });

  it('renders with given state from mocked Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('dispatches an action on input change', () => {
    renderer.act(() => {
      component.root.findByType('input')
        .props
        .onChange({ target: { value: 'sample text' } });
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(setSearchQuery('sample text'));
  });
});
