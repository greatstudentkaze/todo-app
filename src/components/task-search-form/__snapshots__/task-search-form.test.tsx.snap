// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskSearchForm component renders with given state from mocked Redux store 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <svg
    aria-hidden="true"
    className="search-form__search-icon"
    height="14"
    width="14"
  >
    <path
      d="M10.006 8.805h-.633l-.224-.216a5.18 5.18 0 001.257-3.386 5.203 5.203 0 10-5.203 5.203 5.18 5.18 0 003.386-1.257l.216.224v.633L12.807 14 14 12.807l-3.994-4.002zm-4.803 0a3.597 3.597 0 01-3.602-3.602 3.597 3.597 0 013.602-3.602 3.597 3.597 0 013.602 3.602 3.597 3.597 0 01-3.602 3.602z"
    />
  </svg>
  <input
    aria-label="Быстрый поиск"
    className="search-form__input"
    id="search-query"
    name="search-query"
    onChange={[Function]}
    placeholder="Искать"
    type="text"
    value=""
  />
</form>
`;
